cdap-watchdog/src/main/java/co/cask/cdap/logging/write/AvroFileWriter.java

```diff
@@ -193,7 +199,6 @@ private Location createLocation(String pathFragment, long timestamp) throws IOEx
   private AvroFile rotateFile(AvroFile avroFile, LoggingContext loggingContext, long timestamp) throws Exception {
     if (avroFile.getPos() > maxFileSize) {
       LOG.info("Rotating file {}", avroFile.getLocation());
-      flush();
```

Comment:
Closing of the output stream will automatically flush the changes too. The removed flush method was actually flushing all files which is not required at this time.